plugins/router/template/router.go

```diff
@@ -195,10 +203,21 @@ func (r *templateRouter) Commit() error {
 		return err
 	}
 
-	if err := r.reloadRouter(); err != nil {
```

Comment:
This seems more complex than it needs to be - we want

1. a signal that a write to disk and reload is needed
2. a goroutine that watches for that
3. a rate limiter that controls when to actually reload
4. to guarantee commit is thread safe - which the current code doesn't do, because reload and writing to disk can run at the same time

I think we want to keep Commit() the same, but invoke it from a wrapper that returns no error (and have the rest of the code in templateRouter call the wrapper).  The wrapper would log the error if detected.  The important guarantee that has to be preserved is Commit is the *only* method that is allowed to write to disk - I'm not sure that's preserved for SSL keys (so we'd have to check that).  You might have to share a write lock between the commit function and anything that writes to disk.

My instinct here would have been something that looked like this:

```
type RateLimitedFunction struct {
  fn func()
  util.RateLimiter
}
func NewRateLimitedFunction(fn func(), util.RateLimiter) ... {
  ... inits
}

func (RateLimitedFunction) RunUntil(... ch) // like our other controllers

func (RateLimitedFunction) Invoke() // returns nothing

And then have templateRouter initialize the RLF, wrap Commit(), and have another method on templateRouter that is `Commit()` with no error return.  Because this is a deferred error, we shouldn't even bother to handle it here.

Reload isn't really special (enough that the abstraction we are targeting is reload) - instead it's the atomic unit of "invoke this method now or at most X seconds later".